### DISCUSSION TITLE
Deployment problems using node.js

### DISCUSSION THREAD

**Discussion by xiaoxiji1234** [lowdefy/lowdefy#898] [Questions and Help] [answered]
Hello, I deployed the lowdefy project according to the document. After deployment, when I enter the page, I set the openid login to start, but I can't jump to the login page without reporting an error. If it's a development environment, I can jump to login. What's the reason and the solution?

**Comment by SamTolmay** [comment]
Hi @xiaoxiji1234 

What is the error you are seeing? 

Are you seeing the error before you are redirected to the auth provider, on the auth provider hosted page, or after you are redirected back to the Lowdefy app?

It might be a configuration error with the auth provider. You need to configure URLs that your are allowed to redirect to, maybe this is just configured for development but not production?

**Comment by xiaoxiji1234** [comment, marked as answer]
Hi, My program doesn't report any errors, but it doesn't jump to the openid login interface after deployment. It can be used during development

**Reply by SamTolmay** [reply]
It sounds like your OpenID Connect connection parameters are not set correctly. Please make sure you have set the following env variables on your production server:
```
LOWDEFY_SECRET_OPENID_CLIENT_ID
LOWDEFY_SECRET_OPENID_CLIENT_SECRET
LOWDEFY_SECRET_OPENID_DOMAIN
LOWDEFY_SECRET_JWT_SECRET
```

**Reply by xiaoxiji1234** [reply]
Hi, @SamTolmay 
I use the .env configuration file to configure these parameters. The development environment can be adjusted to the login page, but the production environment can't. how can the production environment configure these parameters?
The configuration is as follows:
`
LOWDEFY_SECRET_OPENID_CLIENT_ID = "test"
LOWDEFY_SECRET_OPENID_CLIENT_SECRET = "test"
LOWDEFY_SECRET_OPENID_DOMAIN = "http://localhost:4000/.well-known/openid-configuration"
LOWDEFY_SECRET_JWT_SECRET = "tDZkGun5+CX/p1/asi7q2Gg+ywLQW1d5XZxUCnveTeoFCodXvvSeEJOuw33ip1q/MZEJ5EMds4QLOhUrbPJlVGHERb48Wqs/jxtfr0KrlC/2kcmFWG2TP3wlot5AMo1upxTGCT9EmigHnzUEIfm41thW12mNdauo0fsC8niKSzJsPvCKRDGs89BKm/dRs27W6NWuJmLXUInOG+F6T/ErOfQKWf0FYhZLvW9kOH8E4QpMOfSzb499ahJMpIIjSxRHlGG7eoKuW0NnOPSBRqCEh/FhKIWMCVA7WyiHZQQkRgy0SyQDtJBQkBc9RPjRwuaeocOKaddCartJ0qxuVBPerA=="

`

**Reply by SamTolmay** [reply]
Hi @xiaoxiji1234 

The Lowdefy Node.js server currently does not support `.env` files. Sorry, this wasn't documented very well.

We originally did this because we didn't want people committing .env files to their repositories, but I think it makes sense that they are supported and work like the dev server. 

I have opened an issue to add support for `.env` files in the Node server: #899

You can set the env vars using a shell script. In a file `set_env.sh`:
```sh
export LOWDEFY_SECRET_OPENID_CLIENT_ID = "test"
export LOWDEFY_SECRET_OPENID_CLIENT_SECRET = "test"
# ...
```
Then execute it running `source set_env.sh` before starting your server.

**Reply by xiaoxiji1234** [reply]
Ok, Thank you!
